### PR TITLE
feat(payments): handle currencies properly

### PR DESCRIPTION
**Pull request by paul-nicolas**
Fixes ENG-199

### CODE REVIEW

File: components/payments/cmd/connectors/internal/connectors/bankingcircle/task_fetch_accounts.go [hunk 1]
```diff
@@ -107,32 +109,34 @@ func ingestAccountsBatch(
...
+			DefaultAsset: currency.FormatAsset(supportedCurrenciesWithDecimal, account.Currency),
 			AccountName:  account.AccountDescription,
 			Type:         models.AccountTypeInternal,
 			RawData:      raw,
 		})
 
 		for _, balance := range account.Balances {
+			precision, ok := supportedCurrenciesWithDecimal[balance.Currency]
+			if !ok {
+				precision = 0
```
Comment (gfyrag) [inline components/payments/cmd/connectors/internal/connectors/bankingcircle/task_fetch_accounts.go:121]: default value is already 0

File: components/payments/cmd/connectors/internal/connectors/bankingcircle/task_fetch_payments.go [hunk 2]
```diff
@@ -59,6 +61,8 @@ func taskFetchPayments(
 
 func ingestBatch(
 	ctx context.Context,
+	logger logging.Logger,
```
Comment (gfyrag) [inline components/payments/cmd/connectors/internal/connectors/bankingcircle/task_fetch_payments.go:64]: Not a big deal, but the logger can be retrieved from the context

File: components/payments/cmd/connectors/internal/connectors/bankingcircle/task_fetch_payments.go [hunk 3]
```diff
@@ -73,11 +77,17 @@ func ingestBatch(
...
+			metricsRegistry.ConnectorCurrencyNotSupported().Add(ctx, 1, metric.WithAttributes(connectorAttrs...))
+			continue
+		}
+
 		var amount big.Float
 		amount.SetFloat64(paymentEl.Transfer.Amount.Amount)
-
 		var amountInt big.Int
-		amount.Mul(&amount, big.NewFloat(100)).Int(&amountInt)
+		amount.Mul(&amount, big.NewFloat(math.Pow(10, float64(precision)))).Int(&amountInt)
```
Comment (gfyrag) [inline components/payments/cmd/connectors/internal/connectors/bankingcircle/task_fetch_payments.go:90]: Complicated stuff.
Why not 
amountInt := amount.Mul(&amount, big.NewFloat(math.Pow(10, float64(precision))))

Comment (gfyrag) [reply components/payments/cmd/connectors/internal/connectors/bankingcircle/task_fetch_payments.go:90]: Sorry, the float...

**Review by gfyrag — APPROVED** [review]

**Review by Dav-14 — APPROVED** [review]
I did not see anything weird.

**Pull request merged**